library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -21,18 +21,47 @@ PlatformBridgeFilterConfig::PlatformBridgeFilterConfig(
           Api::External::retrieveApi(proto_config.platform_filter_name()))) {}
 
 PlatformBridgeFilter::PlatformBridgeFilter(PlatformBridgeFilterConfigSharedPtr config)
-    : platform_filter_(config->platform_filter()) {}
+    : platform_filter_(*config->platform_filter()) {
+  // The initialization above sets platform_filter_ to a copy of the struct stored on the config.
+  // In the typical case, this will represent a filter implementation that needs to be intantiated.
+  // static_context will contain the necessary platform-specific mechanism to produce a filter
+  // instance. instance_context will initially be null, but after initialization, set to the
+  // context needed for actual filter invocations.
+
+  // We treat a null initializer as a static implementation.
+  if (platform_filter_.init_filter == nullptr) {
+    // Make static context available to invocations, in lieu of instatiated context.
+    platform_filter_.instance_context = platform_filter_.static_context;
+    return;
+  }
+
+  // Set the instance_context to the result of the initialization call. Cleanup will ultimately
+  // occur during in the onDestroy() invocation below.
+  platform_filter_.instance_context = platform_filter_.init_filter(platform_filter_.static_context);
+
+  ASSERT(platform_filter_.instance_context, "init_filter unsuccessful");
+}
+
+void PlatformBridgeFilter::onDestroy() {
+  // Allow nullptr to act as no-op. Also return if nothing was initialized.
```

Comment:
I liked being able to tell that reaching a nullptr here was a programming mistake.

Reply:
The thing is that as long as Envoy adheres to its contract about filter callbacks, the assertion in `init_filter` will have already been hit.

I decided the extra assertions really amounted to asserting Envoy adheres to its contract, which is better done by Envoy's own coverage.

Reply:
As I alluded above, I also think that it'd be better to treat `nullptr` as a programming error here rather than allowing for static filters